src/sentry/rules/processing/delayed_processing.py

```diff
@@ -288,10 +289,14 @@ def apply_delayed(project_id: int) -> None:
         )
     # Step 7: Fire the rule's actions
     now = timezone.now()
-    # TODO: check rulesnooze table again before firing
     parsed_rulegroup_to_event_data = parse_rulegroup_to_event_data(rulegroup_to_event_data)
-
+    snoozed_rules = RuleSnooze.objects.filter(
```

Comment:
thoughts on moving this condition to step 3? I was thinking it if we filter the snoozed alert rules there (around line 272) it would mean we don't have to process the alerts in the other steps as well.